### ISSUE TITLE
XML Columns and Trigger Creation

### ISSUE THREAD

**Issue by GerardoHess** [Mimetis/Dotmim.Sync#262]
Hi,
     I tried to sync a SQL SERVER database model that holds a table with a XML column using the trigger model. I recieved this error:  _The XML data type cannot be compared or sorted, except when using the IS NULL operator_. I configured the framework to exclude the xml column and it works ok.
    I isolated the bug to the update trigger creation. It seems that IFNULL term doesn't work with XML columns. Have you considered using  COALESCE expression instead? I attached the table script that reproduces the error and a code template for the trigger for your consideration.  Thanks in advance!

[scripts.zip](https://github.com/Mimetis/Dotmim.Sync/files/4524457/scripts.zip)

**Comment by Mimetis** [comment]
You made a mistake in your script.
You want to replace `NULLIF` with `COALESCE`, but the two statements are completely different:

* **COALESCE** evaluates the arguments in order and returns the current value of the first expression that initially doesn't evaluate to NULL
* **NULLIF** returns a null value if the two specified expressions are equal.

Just make a test :
``` sql
Declare @id1 int = 1;
Declare @id2 int = null;

SELECT 
COALESCE(@id1, @id2) as 'COALESCE1',
COALESCE(@id2, @id1) as 'COALESCE2',
NULLIF(@id1, @id2) as 'NULLIF1',
NULLIF(@id2, @id1) as 'NULLIF2',
```
But indeed, maybe you were thinking of replacing `ISNULL` with `COALESCE`

``` sql
COALESCE(NULLIF([d].[ID], [i].[ID]), NULLIF([i].[ID], [d].[ID])) IS NOT NULL
```

But it won't change nothing since the problem is coming from `NULLIF`:

``` sql
Declare @id1 xml = '<head>hello</head>';
Declare @id2 xml = '<head>hello</head>';

SELECT 
COALESCE(@id1, @id2) as 'COALESCE1',
COALESCE(@id2, @id1) as 'COALESCE2',
NULLIF(@id1, @id2) as 'NULLIF1',
NULLIF(@id2, @id1) as 'NULLIF2'
```

The only solution would be to **not** making columns comparison.
That way, it would work, but if one row is updated for *nothing*, like

``` sql
Update Product set Name='Bikes' where Name='Bikes'
```
Then, this row will be sent to the client for nothing...

I made the choice to not supporting **XML** for this reason, unless I found a better solution.
I know, we can have an option saying "*Hey don't use column comparison*", but it adds more complexity than it resolves

**Comment by GerardoHess** [comment]
Hi,

  I would like to first apologize my _rush_ on finding the solution to the problem.  I screw it up!      With your kind explanation I designed another _script_ to decide if an _XML_ column need to be updated. I tested  with various scenarios and it seems now to work OK.  I hope that these _coding pattern_ might help you on deciding on supporting this datatype. I understand that this requirement might be complex to achieve and frameworks are to be kept simple.

Thanks again! I really appreciate your achievements.
[XMLComparison.zip](https://github.com/Mimetis/Dotmim.Sync/files/4526062/XMLComparison.zip)
